android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/NotificationsModule.java

```diff
@@ -50,6 +55,26 @@ public String getName() {
   }
 
   @ReactMethod
+  public void getDevicePushTokenAsync(final Promise promise) {
+    try {
+      InstanceID instanceID = InstanceID.getInstance(this.getReactApplicationContext());
+      String gcmSenderId = Exponent.getInstance().getGCMSenderId();
+      if (gcmSenderId == null || gcmSenderId.length() == 0) {
+        throw new InvalidParameterException("GCM Sender ID is null/empty");
+      }
+      final String token = instanceID.getToken(gcmSenderId, GoogleCloudMessaging.INSTANCE_ID_SCOPE, null);
+      if (token == null) {
+        promise.reject("GCM token has not been set");
+      } else {
+        promise.resolve(token);
+      }
+    } catch (Exception e) {
+      EXL.e(getClass().getSimpleName(), e.getMessage());
```

Comment:
put `private static final String TAG = NotificationsModule.class.getSimpleName();` at the top of the class and then use `TAG` here.